addedIsBlank/isNotBlank_Utils

aditya-32 opened the pull request:
## Description
<!--
Thank a lot for taking time to contribute to Fabric8 <3!

Please provide a description of what your PR does providing a link (if applicable) to the issue it fixes. It is
really helpful for people who would review your code.
-->

## Type of change
<!---
What types of changes does your code introduce? Put an `x` in all the boxes that apply
-->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

<!--
Integration tests (https://github.com/fabric8io/kubernetes-client/tree/master/kubernetes-itests)
Please check integration tests and provide/improve tests if applicable.

Open your PR in Draft mode and verify all of the applicable Checklist items before marking your pull request as ready for review
-->
Added Utils.isBlank() and Utils.isNotBlank() 
fixes #4057

aditya-32 commented:
Hi @manusa @rohanKanojia can you please review the PR

rohanKanojia commented:
@aditya-32 : Could you please link the issue which this PR intends to fix?

aditya-32 commented:
Hi @rohanKanojia  fixes issue: #4057 
updated the description

rohanKanojia commented:
I see. Could you please write some unit tests for this? You would also need to check and update other places where we're using `== null`/ `!= null` etc checks.

aditya-32 commented:
Hi @rohanKanojia  can you please review it , i have added unit test cases and replaced instances of ==null and != null with the use of Utils.isBlank() and Utils.isNotBlank() . Also please suggest any changes if needed or any where you see i can replace null check with utils.isBlank().

aditya-32 commented:
@rohanKanojia @manusa  can you please review the PR

aditya-32 commented:
any update on the PR ?
can we review this

shawkins reviewed (changes requested):
The application of isBlank/isNotBlank here is overly broad.  It should initially be applied only to DSL entry points for user input - withName, lockResourceVersion, inNamespace, etc.

The intention was to consolidate where we were inconsistently checking for the empty case.

While trimming to look for a string that contains only whitespace will be valid for user input, it will not be valid in general for places like https://github.com/fabric8io/kubernetes-client/pull/5066/files#diff-890933cbcc1b567c6f5164f77ec5ede516a4422549bfa6e917f319c67d863025R256 - where you could be replacing whitespace.

If we want to go further with validation many of these places will need to be valid kubernetes identifiers, there is already the KubernetesResourceUtil.isValidName - although glancing at that logic it may not be quite right either as it doesn't look to understand annotation keys.
